src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -375,6 +380,132 @@ suggestSignature isQuickFix Diagnostic{_range=_range@Range{..},..}
 
 suggestSignature _ _ = []
 
+findMissingConstraint :: T.Text -> Maybe T.Text
+findMissingConstraint t =
+    let regex = "(No instance for|Could not deduce) \\((.+)\\) arising from a use of"
+     in matchRegex t regex <&> last
+
+normalizeConstraints :: T.Text -> T.Text -> T.Text
+normalizeConstraints existingConstraints constraint =
+  let constraintsInit = if "(" `T.isPrefixOf` existingConstraints
+                           then T.dropEnd 1 existingConstraints
+                           else "(" <> existingConstraints
+   in constraintsInit <> ", " <> constraint <> ")"
+
+-- | Suggests a constraint for an instance declaration for which a constraint is missing.
+suggestInstanceConstraint :: Maybe T.Text -> Diagnostic -> [(T.Text, [TextEdit])]
+suggestInstanceConstraint contents Diagnostic {..}
+-- Suggests a constraint for an instance declaration with no existing constraints.
+-- • No instance for (Eq a) arising from a use of ‘==’
+--   Possible fix: add (Eq a) to the context of the instance declaration
+-- • In the expression: x == y
+--   In an equation for ‘==’: (Wrap x) == (Wrap y) = x == y
+--   In the instance declaration for ‘Eq (Wrap a)’
```

Comment:
```suggestion
-- | Suggests a constraint for an instance declaration for which a constraint is missing.
--
-- Suggests a constraint for an instance declaration with no existing constraints.
-- • No instance for (Eq a) arising from a use of ‘==’
--   Possible fix: add (Eq a) to the context of the instance declaration
-- • In the expression: x == y
--   In an equation for ‘==’: (Wrap x) == (Wrap y) = x == y
--   In the instance declaration for ‘Eq (Wrap a)’
suggestInstanceConstraint :: Maybe T.Text -> Diagnostic -> [(T.Text, [TextEdit])]
suggestInstanceConstraint contents Diagnostic {..}
```

I mean this. This way, on hover on this function, you can see all comments.
But if you think that is a bad idea, that is fine, too.

Reply:
I see. The comment you moved refers only to the branch of the pattern matching, so it's relevant to an implementation detail. Besides, the first line would be kind of redundant, wouldn't it? There is also another comment similar to this one for the branch of the pattern match right below. If we moving both of them out of that context they would loose their meaning, since they strictly refer to the branches. What do you think?

Reply:
> the first line would be kind of redundant, wouldn't it?

Yeah, but I dont think this is really bad.

> There is also another comment similar to this one 

Yeah, I would move that, too.

> What do you think?

Your concern seems valid to me. *Personally*, I would like it to have it available as haddock comments. *But*, if you think it is too much of an implementation detail to actually show in the documentation, you are probably right. So, whatever feels right to you!